apps/wallet/src/ui/app/components/accounts/ImportRecoveryPhraseForm.tsx

```diff
@@ -0,0 +1,200 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useZodForm } from '@mysten/core';
+import { z } from 'Zod';
+import { type SubmitHandler } from 'react-hook-form';
+import { useNavigate } from 'react-router-dom';
+import Alert from '../alert';
+import { Button } from '_app/shared/ButtonUI';
+import { normalizeMnemonics, validateMnemonics } from '_src/shared/utils/bip39';
+import { PasswordInput } from '_src/ui/app/shared/forms/controls/PasswordInput';
+import { Text } from '_src/ui/app/shared/text';
+
+const RECOVERY_PHRASE_WORD_COUNT = 12;
+
+const formSchema = z.object({
+	recoveryPhrase: z
+		.array(z.string().trim())
+		.length(RECOVERY_PHRASE_WORD_COUNT)
+		.transform((recoveryPhrase) => normalizeMnemonics(recoveryPhrase.join(' ')).split(' '))
+		.refine((recoveryPhrase) => validateMnemonics(recoveryPhrase.join(' ')), {
+			message: 'Recovery Passphrase is invalid',
+		}),
+});
+
+type FormValues = z.infer<typeof formSchema>;
+
+type ImportRecoveryPhraseFormProps = {
+	submitButtonText: string;
+	cancelButtonText: string;
+	useFieldSet?: boolean;
```

Comment:
I don't like having this prop... but I'm not sure how to best handle the fact that these forms are 98% the same except for the "Enter your 12-word Recovery Phrase" text positioning 🤔 

<img width="223" alt="image" src="https://github.com/MystenLabs/sui/assets/7453188/84f33bd0-d434-4f4e-b202-92e9d5431a94">


<img width="402" alt="image" src="https://github.com/MystenLabs/sui/assets/7453188/a1e70d45-f61f-4373-98a7-39da2c7c652f">